src/components/createContainer.jsx

```diff
@@ -32,41 +34,47 @@ export default function createContainer(DecoratedComponent, specs) {
     static DecoratedComponent = DecoratedComponent
 
     static contextTypes = {
-      renderLoading: PropTypes.func.isRequired,
-      adaptor: adaptorShape.isRequired,
+      adrenaline: PropTypes.shape({
+          renderLoading: PropTypes.func.isRequired,
+          adaptor: adaptorShape.isRequired
+      }).isRequired
+    }
+
+    static childContextTypes = {
+        adrenaline: containerShape.isRequired
     }
 
     constructor(props, context) {
       super(props, context);
-
       this.isNotMounted = true;
+      this.state = {};
+    }
 
-      const { adaptor } = context;
-      //DecoratedComponent.prototype.shouldComponentUpdate = adaptor.shouldComponentUpdate;
-
-      const initialArgs = !!specs.args ? specs.args(props) : {};
-
-      this.state = {
-        data: {},
-        args: initialArgs,
```

Comment:
I don't think we can ever set args without the corresponding data since they're semantically coupled. I've moved this to ```componentDidMount()``` to kick off the initial load, but I think it would be safe to do this in ```componentWillMount```